app/src/main/java/com/example/anfcodetest/MainActivity.kt

```diff
@@ -1,55 +1,36 @@
 package com.example.anfcodetest
 
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
 import androidx.recyclerview.widget.LinearLayoutManager
-import com.fasterxml.jackson.module.kotlin.*
+import com.hannesdorfmann.mosby3.mvp.MvpActivity
 import kotlinx.android.synthetic.main.activity_main.*
-import okhttp3.*
-import java.io.IOException
 
-val url = "https://www.abercrombie.com/anf/nativeapp/qa/codetest/codeTest_exploreData.json"
-
-class MainActivity : AppCompatActivity(), MainContract.View {
+class MainActivity : MvpActivity<MainContract.View, MainContract.Presenter>(), MainContract.View {
```

Comment:
You could move this to an MVP activity following the Mosby examples here: [Mosby MVP](http://hannesdorfmann.com/mosby/getting-started/)

This allows us to tie our presenter to our view and call those methods directly. This way, you don't need the JSON method in this class anymore!